hybridq/architecture/utils.py

```diff
@@ -0,0 +1,152 @@
+"""
+Author: Salvatore Mandra (salvatore.mandra@nasa.gov)
+
+Copyright © 2021, United States Government, as represented by the Administrator
+of the National Aeronautics and Space Administration. All rights reserved.
+
+The HybridQ: A Hybrid Simulator for Quantum Circuits platform is licensed under
+the Apache License, Version 2.0 (the "License"); you may not use this file
+except in compliance with the License. You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+Types
+-----
+**`Qubit`**: `tuple[int, int]`
+
+**`QpuLayout`**: `list[Qubit]`
+
+**`Coupling`**: `tuple[Qubit, Qubit]`
+"""
+
+from __future__ import annotations
+from typing import List, Tuple, Callable
+
+__all__ = ['get_layout']
+
+# Define Qubit type
+Qubit = Tuple[int, int]
+
+# Define Coupling
+Coupling = Tuple[Qubit, Qubit]
+
+# Define QpuLayout
+QpuLayout = List[Qubit]
+
+
+def get_layout(layout: str) -> tuple[list[Qubit], list[Coupling]]:
```

Comment:
How does this relate to the architecture-specific get_layout, above

Reply:
To avoid this confusion, I did the following changes:

* `hybridq.architecture.utils.get_layout` -> `hybridq.architecture.utils.get_layout_from_drawing`: This function returns `qubits` and `couplings` accordingly to a given drawing (see: `tutorials/qpu_layouts.ipynb`)

* `hybridq.architecture.sycamore.get_layers`: It's a function specifically designed for Sycamore-like QPU that returns a subset of couplings (layer) given a set of qubits (layout), accordingly to our paper [Nature 574 (7779), 505-510].

Hope this helps to clarify the difference.